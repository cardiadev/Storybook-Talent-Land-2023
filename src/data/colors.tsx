import { IColor } from "../@interfaces";

export const colors: IColor[] = [
  {
    bgClass: "bg-gray-25",
    code: 25,
    hex: "#FCFCFD",
  },
  {
    bgClass: "bg-gray-50",
    code: 50,
    hex: "#F9FAFB",
  },
  {
    bgClass: "bg-gray-100",
    code: 100,
    hex: "#F2F4F7",
  },
  {
    bgClass: "bg-gray-200",
    code: 200,
    hex: "#E4E7EC",
  },
  {
    bgClass: "bg-gray-300",
    code: 300,
    hex: "#D0D5DD",
  },
  {
    bgClass: "bg-gray-400",
    code: 400,
    hex: "#98A2B3",
  },
  {
    bgClass: "bg-gray-500",
    code: 500,
    hex: "#667085",
  },
  {
    bgClass: "bg-gray-600",
    code: 600,
    hex: "#475467",
  },
  {
    bgClass: "bg-gray-700",
    code: 700,
    hex: "#344054",
  },
  {
    bgClass: "bg-gray-800",
    code: 800,
    hex: "#1D2939",
  },
  {
    bgClass: "bg-gray-900",
    code: 900,
    hex: "#101828",
  },
  {
    bgClass: "bg-primary-25",
    code: 25,
    hex: "#F5FAFF",
  },
  {
    bgClass: "bg-primary-50",
    code: 50,
    hex: "#EFF8FF",
  },
  {
    bgClass: "bg-primary-100",
    code: 100,
    hex: "#D1E9FF",
  },
  {
    bgClass: "bg-primary-200",
    code: 200,
    hex: "#B2DDFF",
  },
  {
    bgClass: "bg-primary-300",
    code: 300,
    hex: "#84CAFF",
  },
  {
    bgClass: "bg-primary-400",
    code: 400,
    hex: "#53B1FD",
  },
  {
    bgClass: "bg-primary-500",
    code: 500,
    hex: "#2E90FA",
  },
  {
    bgClass: "bg-primary-600",
    code: 600,
    hex: "#1570EF",
  },
  {
    bgClass: "bg-primary-700",
    code: 700,
    hex: "#175CD3",
  },
  {
    bgClass: "bg-primary-800",
    code: 800,
    hex: "#1849A9",
  },
  {
    bgClass: "bg-primary-900",
    code: 900,
    hex: "#194185",
  },
  {
    bgClass: "bg-error-25",
    code: 25,
    hex: "#FFFBFA",
  },
  {
    bgClass: "bg-error-50",
    code: 50,
    hex: "#FEF3F2",
  },
  {
    bgClass: "bg-error-100",
    code: 100,
    hex: "#FEE4E2",
  },
  {
    bgClass: "bg-error-200",
    code: 200,
    hex: "#FECDCA",
  },
  {
    bgClass: "bg-error-300",
    code: 300,
    hex: "#FDA29B",
  },
  {
    bgClass: "bg-error-400",
    code: 400,
    hex: "#F97066",
  },
  {
    bgClass: "bg-error-500",
    code: 500,
    hex: "#F04438",
  },
  {
    bgClass: "bg-error-600",
    code: 600,
    hex: "#D92D20",
  },
  {
    bgClass: "bg-error-700",
    code: 700,
    hex: "#B42318",
  },
  {
    bgClass: "bg-error-800",
    code: 800,
    hex: "#912018",
  },
  {
    bgClass: "bg-error-900",
    code: 900,
    hex: "#7A271A",
  },
  {
    bgClass: "bg-warning-25",
    code: 25,
    hex: "#FFFCF5",
  },
  {
    bgClass: "bg-warning-50",
    code: 50,
    hex: "#FFFAEB",
  },
  {
    bgClass: "bg-warning-100",
    code: 100,
    hex: "#FEF0C7",
  },
  {
    bgClass: "bg-warning-200",
    code: 200,
    hex: "#FEDF89",
  },
  {
    bgClass: "bg-warning-300",
    code: 300,
    hex: "#FEC84B",
  },
  {
    bgClass: "bg-warning-400",
    code: 400,
    hex: "#FDB022",
  },
  {
    bgClass: "bg-warning-500",
    code: 500,
    hex: "#F79009",
  },
  {
    bgClass: "bg-warning-600",
    code: 600,
    hex: "#DC6803",
  },
  {
    bgClass: "bg-warning-700",
    code: 700,
    hex: "#B54708",
  },
  {
    bgClass: "bg-warning-800",
    code: 800,
    hex: "#93370D",
  },
  {
    bgClass: "bg-warning-900",
    code: 900,
    hex: "#7A2E0E",
  },
  {
    bgClass: "bg-success-25",
    code: 25,
    hex: "#F6FEF9",
  },
  {
    bgClass: "bg-success-50",
    code: 50,
    hex: "#ECFDF3",
  },
  {
    bgClass: "bg-success-100",
    code: 100,
    hex: "#D1FADF",
  },
  {
    bgClass: "bg-success-200",
    code: 200,
    hex: "#A6F4C5",
  },
  {
    bgClass: "bg-success-300",
    code: 300,
    hex: "#6CE9A6",
  },
  {
    bgClass: "bg-success-400",
    code: 400,
    hex: "#32D583",
  },
  {
    bgClass: "bg-success-500",
    code: 500,
    hex: "#12B76A",
  },
  {
    bgClass: "bg-success-600",
    code: 600,
    hex: "#039855",
  },
  {
    bgClass: "bg-success-700",
    code: 700,
    hex: "#027A48",
  },
  {
    bgClass: "bg-success-800",
    code: 800,
    hex: "#05603A",
  },
  {
    bgClass: "bg-success-900",
    code: 900,
    hex: "#054F31",
  },
];
