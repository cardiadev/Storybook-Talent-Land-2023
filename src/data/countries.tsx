import React from "react";
import { IOption } from "../@interfaces";
import { CountryISO, images } from "./images";

interface ICountry {
  city: string;
  countryISO: CountryISO;
}

const countriesData: ICountry[] = [
  {
    city: "Amsterdam",
    countryISO: "NL",
  },
  {
    city: "Rome",
    countryISO: "IT",
  },
  {
    city: "Berlin",
    countryISO: "DE",
  },
  {
    city: "Paris",
    countryISO: "FR",
  },
  {
    city: "Washington",
    countryISO: "US",
  },
  {
    city: "Moscow",
    countryISO: "RU",
  },
  {
    city: "Brussels",
    countryISO: "BE",
  },
];

export const countries: IOption[] = countriesData.map((country) => ({
  value: country.countryISO,
  label: (
    <>
      <img
        src={images[country.countryISO]}
        alt={country.countryISO.toLowerCase()}
        className="w-5 h-5 mr-2"
      />{" "}
      {`${country.city}, ${country.countryISO}`}
    </>
  ),
}));
