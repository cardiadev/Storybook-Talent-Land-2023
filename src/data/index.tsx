export * from "./colors";
export * from "./countries";
export * from "./dates";
export * from "./prices";
export * from "./rentals";
export * from "./navItems";
export * from "./options";
export * from "./figma";
