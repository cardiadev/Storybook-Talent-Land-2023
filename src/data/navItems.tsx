import React from "react";
import { INavItem } from "../@interfaces";
import {
  FiArrowLeft,
  FiBarChart2,
  FiCheckSquare,
  FiHome,
  FiLayers,
  FiSettings,
  FiFlag,
  FiMapPin,
} from "react-icons/fi";

export const navItemsTop: INavItem[] = [
  {
    order: 0,
    path: "home",
    label: "Home",
    icon: <FiHome />,
  },
  {
    order: 1,
    path: "dashboard",
    label: "Dashboard",
    icon: <FiBarChart2 />,
    subItems: [
      {
        order: 0,
        path: "analytics",
        label: "Analytics",
      },
      {
        order: 1,
        path: "apps",
        label: "Apps",
      },
      {
        order: 2,
        path: "learn",
        label: "Learn",
      },
    ],
  },
  {
    order: 2,
    path: "projects",
    label: "Projects",
    icon: <FiLayers />,
  },
  {
    order: 3,
    path: "tasks",
    label: "Tasks",
    icon: <FiCheckSquare />,
  },
  {
    order: 4,
    path: "reporting",
    label: "Reporting",
    icon: <FiFlag />,
  },
  {
    order: 5,
    path: "rentals",
    label: "Rentals",
    icon: <FiMapPin />,
    subItems: [
      {
        order: 0,
        path: "all",
        label: "All rentals",
      },
      {
        order: 1,
        path: "popular",
        label: "Popular",
      },
      {
        order: 2,
        path: "recent",
        label: "Recently added",
      },
    ],
  },
];

export const navItemsBottom: INavItem[] = [
  {
    order: 0,
    path: "settings",
    label: "Settings",
    icon: <FiSettings />,
  },
  {
    order: 1,
    label: "Close Navbar",
    icon: <FiArrowLeft />,
    toggleSidebar: true,
  },
];
