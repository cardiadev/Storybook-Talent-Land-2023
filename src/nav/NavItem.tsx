import classNames from "classnames";
import React from "react";

import { FiChevronDown } from "react-icons/fi";
import { Typography } from "../Typography";
import { INavItem } from "../@interfaces";
import { SubNavItem } from "./SubNavItem";

import { Popover } from "@headlessui/react";

interface INavItemProps {
  item: INavItem;
  isActive: boolean;
  activeSubNavItem: string;
  open: boolean;
  openDropdown: boolean;
  onClick: (item: INavItem, subNavPath?: string) => void;
}

export const NavItem = ({
  item,
  isActive,
  activeSubNavItem,
  open,
  openDropdown,
  onClick,
}: INavItemProps) => (
  <Popover key={item.label} className="relative">
    <Popover.Button as={React.Fragment}>
      <li
        className={classNames("navItem group", {
          "px-4": open,
          "ml-2 px-2 w-10": !open,
          "bg-primary-50 dark:bg-gray-100 dark:bg-opacity-10": isActive,
        })}
        onClick={() => onClick(item)}
      >
        <item.icon.type
          size={24}
          className={classNames(
            "group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out",
            {
              "-rotate-180": item.toggleSidebar && !open,
              "text-gray-900": !isActive,
              "text-primary-600 dark:text-white": isActive,
            },
          )}
          {...item.icon.props}
        />
        <>
          <Typography
            variant="md"
            customWeight="medium"
            className={classNames("flex-grow ml-3", {
              "opacity-0 hidden": !open,
            })}
            customColor={classNames(
              "group-hover:text-primary-600 dark:group-hover:text-white",
              {
                "text-gray-800 dark:text-white": !isActive,
                "text-primary-600 dark:text-white": isActive,
              },
            )}
          >
            {item.label}
          </Typography>
          {item.subItems && item.subItems.length > 0 ? (
            <FiChevronDown
              size={20}
              className={classNames(
                "group-hover:text-primary-600 dark:group-hover:text-white transform duration-100 ease-out",
                {
                  "opacity-0 hidden": !open,
                  "text-gray-400": !isActive,
                  "text-primary-600 dark:text-white": isActive,
                  "-rotate-180": openDropdown,
                },
              )}
            />
          ) : null}
        </>
      </li>
    </Popover.Button>

    {openDropdown && open && item.subItems && item.subItems.length > 0 ? (
      <div className="my-2 space-y-1">
        {item.subItems.map((subItem) => (
          <SubNavItem
            key={subItem.label}
            isActive={activeSubNavItem === subItem.label}
            subItem={subItem}
            onClick={() => onClick(item, subItem.label)}
          />
        ))}
      </div>
    ) : null}

    {!open && item.subItems && item.subItems.length > 0 ? (
      <Popover.Panel className="absolute z-50 px-4 py-3 ml-5 space-y-1 bg-white rounded-lg shadow-md -top-3 left-19 w-58 dark:bg-gray-900">
        {item.subItems.map((subItem) => (
          <SubNavItem
            key={subItem.label}
            inDropdown
            isActive={activeSubNavItem === subItem.label}
            subItem={subItem}
            onClick={() => onClick(item, subItem.label)}
          />
        ))}
      </Popover.Panel>
    ) : null}
  </Popover>
);
