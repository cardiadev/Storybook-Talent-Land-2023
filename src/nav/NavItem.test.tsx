import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { NavItem } from "./NavItem";
import { navItemsTop, navItemsBottom } from "../data/navItems";

afterEach(cleanup);

// TODO test clicking on NavItem
// TODO test clicking on SubNavItem

const setup = () => {
  const navItems = [...navItemsTop, ...navItemsBottom];
  const activeNavItem = "rentals";
  const activeSubNavItem = "all";
  const openDropdowns = ["rentals"];

  return {
    navItems,
    activeNavItem,
    activeSubNavItem,
    openDropdowns,
  };
};

describe("NavItem", () => {
  it("Renders NavItems with open SideNav correctly", () => {
    const { navItems, activeNavItem, activeSubNavItem, openDropdowns } =
      setup();

    const { asFragment } = render(
      <>
        {navItems.map((item) => (
          <NavItem
            key={item.label}
            item={item}
            isActive={activeNavItem === item.label}
            activeSubNavItem={activeSubNavItem}
            open
            openDropdown={openDropdowns.includes(item.label)}
            onClick={() => {}}
          />
        ))}
      </>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders NavItems with closed SideNav correctly", () => {
    const { navItems, activeNavItem, activeSubNavItem, openDropdowns } =
      setup();

    const { asFragment } = render(
      <>
        {navItems.map((item) => (
          <NavItem
            key={item.label}
            item={item}
            isActive={activeNavItem === item.label}
            activeSubNavItem={activeSubNavItem}
            open={false}
            openDropdown={openDropdowns.includes(item.label)}
            onClick={() => {}}
          />
        ))}
      </>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
