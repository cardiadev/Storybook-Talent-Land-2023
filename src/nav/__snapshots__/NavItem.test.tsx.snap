// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Renders NavItems with closed SideNav correctly 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group ml-2 px-2 w-10"
      id="headlessui-popover-button-17"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
        />
        <polyline
          points="9 22 9 12 15 12 15 22"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 opacity-0 hidden group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Home
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group ml-2 px-2 w-10"
      id="headlessui-popover-button-19"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="18"
          y1="20"
          y2="10"
        />
        <line
          x1="12"
          x2="12"
          y1="20"
          y2="4"
        />
        <line
          x1="6"
          x2="6"
          y1="20"
          y2="14"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 opacity-0 hidden group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Dashboard
      </p>
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white transform duration-100 ease-out opacity-0 hidden text-gray-400"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="6 9 12 15 18 9"
        />
      </svg>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group ml-2 px-2 w-10"
      id="headlessui-popover-button-21"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="12 2 2 7 12 12 22 7 12 2"
        />
        <polyline
          points="2 17 12 22 22 17"
        />
        <polyline
          points="2 12 12 17 22 12"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 opacity-0 hidden group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Projects
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group ml-2 px-2 w-10"
      id="headlessui-popover-button-23"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 11 12 14 22 4"
        />
        <path
          d="M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 opacity-0 hidden group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Tasks
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group ml-2 px-2 w-10"
      id="headlessui-popover-button-25"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 15s1-1 4-1 5 2 8 2 4-1 4-1V3s-1 1-4 1-5-2-8-2-4 1-4 1z"
        />
        <line
          x1="4"
          x2="4"
          y1="22"
          y2="15"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 opacity-0 hidden group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Reporting
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group ml-2 px-2 w-10"
      id="headlessui-popover-button-27"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
        />
        <circle
          cx="12"
          cy="10"
          r="3"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 opacity-0 hidden group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Rentals
      </p>
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white transform duration-100 ease-out opacity-0 hidden text-gray-400"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="6 9 12 15 18 9"
        />
      </svg>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group ml-2 px-2 w-10"
      id="headlessui-popover-button-29"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="3"
        />
        <path
          d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 opacity-0 hidden group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Settings
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group ml-2 px-2 w-10"
      id="headlessui-popover-button-31"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out -rotate-180 text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="19"
          x2="5"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 19 5 12 12 5"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 opacity-0 hidden group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Close Navbar
      </p>
    </li>
  </div>
</DocumentFragment>
`;

exports[`NavItem Renders NavItems with open SideNav correctly 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group px-4"
      id="headlessui-popover-button-1"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
        />
        <polyline
          points="9 22 9 12 15 12 15 22"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Home
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group px-4"
      id="headlessui-popover-button-3"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="18"
          y1="20"
          y2="10"
        />
        <line
          x1="12"
          x2="12"
          y1="20"
          y2="4"
        />
        <line
          x1="6"
          x2="6"
          y1="20"
          y2="14"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Dashboard
      </p>
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white transform duration-100 ease-out text-gray-400"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="6 9 12 15 18 9"
        />
      </svg>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group px-4"
      id="headlessui-popover-button-5"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="12 2 2 7 12 12 22 7 12 2"
        />
        <polyline
          points="2 17 12 22 22 17"
        />
        <polyline
          points="2 12 12 17 22 12"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Projects
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group px-4"
      id="headlessui-popover-button-7"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 11 12 14 22 4"
        />
        <path
          d="M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Tasks
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group px-4"
      id="headlessui-popover-button-9"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 15s1-1 4-1 5 2 8 2 4-1 4-1V3s-1 1-4 1-5-2-8-2-4 1-4 1z"
        />
        <line
          x1="4"
          x2="4"
          y1="22"
          y2="15"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Reporting
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group px-4"
      id="headlessui-popover-button-11"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
        />
        <circle
          cx="12"
          cy="10"
          r="3"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Rentals
      </p>
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white transform duration-100 ease-out text-gray-400"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="6 9 12 15 18 9"
        />
      </svg>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group px-4"
      id="headlessui-popover-button-13"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="3"
        />
        <path
          d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Settings
      </p>
    </li>
  </div>
  <div
    class="relative"
  >
    <li
      aria-expanded="false"
      class="navItem group px-4"
      id="headlessui-popover-button-15"
    >
      <svg
        class="group-hover:text-primary-600 dark:group-hover:text-white dark:text-white transform duration-100 ease-out text-gray-900"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="19"
          x2="5"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 19 5 12 12 5"
        />
      </svg>
      <p
        class="text-md font-medium flex-grow ml-3 group-hover:text-primary-600 dark:group-hover:text-white text-gray-800 dark:text-white"
      >
        Close Navbar
      </p>
    </li>
  </div>
</DocumentFragment>
`;
